pkg/ipam/ipallocator/allocator.go

```diff
@@ -218,6 +283,60 @@ func (ma MultiIPAllocator) AllocateNext() (net.IP, error) {
 	return nil, fmt.Errorf("cannot allocate IP in any range")
 }
 
+// IP space starting from startIP, if specified. If continuus chunk is not
+// available, other available IPs will be allocated, possibly from different
+// CIDRs/ranges.
+// With bestEffort=true, AllocateChunk will succeed even if not enough IPs were
+// available for allocation.
+func (ma MultiIPAllocator) AllocateChunk(startIP net.IP, chunkSize int, bestEffort bool) ([]net.IP, error) {
+	var ips []net.IP
+
+	if !bestEffort && (chunkSize > ma.Free()) {
+		return nil, fmt.Errorf("Not enough IPs to reserve chunk of size %d", chunkSize)
+	}
+
+	var firstAllocator *SingleIPAllocator
+	// Find the allocator to start from
+	if startIP != nil {
+		for _, a := range ma {
+			if a.Has(startIP) {
+				firstAllocator = a
+			}
+		}
+	}
+
+	// Try our best to allocate from startIP onwards
+	if firstAllocator != nil {
+		if startIPs, _ := firstAllocator.AllocateChunk(startIP, chunkSize, true); startIPs != nil {
+			ips = append(ips, startIPs...)
+			if len(ips) == chunkSize {
+				return ips, nil
+			}
+		}
+	}
+
+	// Not enough IPs allocated - proceed to other allocators
+	for _, a := range ma {
+		size := chunkSize - len(ips)
+		if nextIPs, _ := a.AllocateChunk(nil, size, true); nextIPs != nil {
```

Comment:
Do we need to add a log here? Or explain the behavior in doc?

Reply:
Sure, I can add log message, but I'd like to hear some feedback regarding whether non-continuous chunk allocation is needed. If not, behavior here can be simplified.